posts/draft.md

```diff
@@ -0,0 +1,174 @@
+
+```python
+def default_parser(config: MdParserConfig) -> MarkdownIt:
+    """Return the default parser configuration for MyST"""
+    renderer_cls: Callable[[MarkdownIt], RendererProtocol]
+
+    if config.renderer == "sphinx":
+        from myst_parser.sphinx_renderer import SphinxRenderer
+
+        renderer_cls = SphinxRenderer
+    elif config.renderer == "html":
+        renderer_cls = RendererHTML
+    elif config.renderer == "docutils":
+        from myst_parser.docutils_renderer import DocutilsRenderer
+
+        renderer_cls = DocutilsRenderer
+    else:
+        raise ValueError("unknown renderer type: {0}".format(config.renderer))
+```
+
+Well, this is quite interesting! We have a `SphinxRenderer` and `DocutilsRenderer`. You
+can learn about those [here](https://myst-parser.readthedocs.io/en/latest/api/renderers.html#).
+Those are an integral part of the Myst-Parser project. But when you look at the `html`
+renderer, that [RendererHTML] class (https://markdown-it-py.readthedocs.io/en/latest/api/markdown_it.renderer.html#markdown_it.renderer.RendererHTML)
+is coming from another EBP project called markdown-it-py. This is a Python port of the
+well-used [markdown-it](https://github.com/markdown-it/markdown-it), plus some of itsassociated plugins.
+
+If you indeed keep looking at the `default_parser` implementation, you will find the
+MyST-Parser is actually a collection of markdown-it-py plugins:
+
+```python
+    md = (
+        MarkdownIt("commonmark", renderer_cls=renderer_cls)
+        .enable("table")
+        .use(front_matter_plugin)
+        .use(myst_block_plugin)
+        .use(myst_role_plugin)
+        .use(footnote_plugin)
+        .use(wordcount_plugin, per_minute=config.words_per_minute)
+        .disable("footnote_inline")
+        # disable this for now, because it need a new implementation in the renderer
+        .disable("footnote_tail")
+    )
+```
+
+Notice there are some MyST specific plugins such as the `myst_block_plugin` and the
+`myst_role_plugin`. The implementation is more complex, of course, I am just extracting
+the core idea ;-)
+
+So, it seems the main idea is to have a `MystParser`, a Sphinx parser, that parses stuff
+into a token stream using a set of general and some specific markdown-it-py plugins,
+creates a Docutils objects-based representation of the token stream and render those
+objects using the Sphinx machinery.
+
+In this model, it actually makes sense to see roles and directives supported when you go
+through the Docutils/Sphinx path (those concepts are core-supported in the
+Docutils/Sphinx ecosystem) and those unsupported when you use the MyST Python API which
+does not understand how to parse nor render these concepts as we saw previously.
+
+# Overcome the limitations (don't worry I will stop here... for now)
+
+OK, this is long enough for now, the next post will deep dive a little bit more in the
+Myst-parser and the markdown-it-py codebases in the context of potentially supporting
```

Comment:
```suggestion
MyST-parser and the markdown-it-py codebases in the context of potentially supporting
```

Reply:
I think it should be either `myst` (all lowercase) or `MyST`, but not intermixed (`Myst`)

Reply:
I totally agree!